### PR TITLE
New token validation path: Renamed result types

### PR DESCRIPTION
**Pull request by iNinja**
# New token validation path: Renamed result types

Updated the type names for the new token validation path to prevent future clashes with the type name "Result".

ValidationResult -> **ValidatedToken**
ExceptionDetails -> **ValidationError**
Result -> **ValidationResult**

Validators and delegates now return **ValidationResult<TResult>**.
`ValidateTokenAsync` now returns **ValidationResult\<ValidatedToken\>**
Failed validations now return a **ValidationError** within `UnwrapError()`

Part of #2711

### CODE REVIEW

File: src/Microsoft.IdentityModel.Tokens/Validation/Results/ValidationResult.cs [hunk 1]
```diff
@@ -2,166 +2,157 @@
...
+        /// <returns>The wrapped error value.</returns>
+        /// <remarks>This method is only valid if the result type is unsuccessful.</remarks>
+        /// <exception cref="InvalidOperationException">Thrown if attempted to unwrap an error from a successful result.</exception>
+        public ValidationError UnwrapError() => IsSuccess ? throw new InvalidOperationException("Cannot unwrap success result") : _error!;
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="obj"></param>
+        /// <returns></returns>
```
Comment (keegan-caruso) [inline src/Microsoft.IdentityModel.Tokens/Validation/Results/ValidationResult.cs:85]: Can probably just `<inheritdoc/>` this

Comment (keegan-caruso) [reply src/Microsoft.IdentityModel.Tokens/Validation/Results/ValidationResult.cs:85]: Same for GetHashCode

File: src/Microsoft.IdentityModel.Tokens/Validation/Results/ValidatedToken.cs [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+        }
+
+        /// <summary>
+        /// Logs the validation result.
+        /// </summary>
+#pragma warning disable CA1822 // Mark members as static
+        public void Log()
+#pragma warning restore CA1822 // Mark members as static
+        {
+            // TODO - Do we need this, how will it work?
```
Comment (keegan-caruso) [inline src/Microsoft.IdentityModel.Tokens/Validation/Results/ValidatedToken.cs:41]: I'd avoid adding this method until decide on how it will work.

**Comment by iNinja** [comment]
Git did a whole mess due to the fact that I renamed a different file to the name that a previous one had. So now it is comparing OldA against newB.

This PR made no changes to the renamed types, only to the references in other files.

**Review by keegan-caruso — APPROVED** [review]

**Review by brentschmaltz — APPROVED** [review]

**Pull request merged**